client/src/components/models/DatasetSelector.tsx

```diff
@@ -0,0 +1,108 @@
+import LoadingIndicator from 'components/LoadingIndicator';
+import Prose from 'components/Prose';
+import {useAuth} from 'contexts/auth';
+import {getDatasets} from 'lib/api/datasets';
+import Link from 'next/link';
+import React, {useEffect, useState} from 'react';
+import {Dataset} from 'types/Dataset';
+import {DownloadState} from 'types/LoadingStates';
+
+type Props = {
+  currentDataset: Dataset | null;
+  selectDataset(dataset: Dataset): void;
+};
+
+export default function DatasetSelector({
+  currentDataset,
+  selectDataset,
+}: Props) {
+  const {user} = useAuth();
+  const [loadingState, setLoadingState] = useState<DownloadState>('waiting');
+  const [datasets, setDatasets] = useState<Dataset[]>([]);
+
+  useEffect(() => {
+    if (user) {
+      setLoadingState('downloading');
+      try {
+        getUserDatasets();
+        setLoadingState('completed');
+      } catch {
+        setLoadingState('error');
+      }
+    }
+  }, [user]);
+
+  async function getUserDatasets() {
+    const userDatasets = await getDatasets(user!);
+    setDatasets(userDatasets);
+  }
+
+  const chooseDataset = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    if (e.target.value === '') return;
+
+    const dataset = datasets.find(
+      (dataset: Dataset) => dataset.name === e.target.value
+    );
+    selectDataset(dataset!);
+  };
+
+  // Loading state
+  if (loadingState === 'waiting' || loadingState === 'downloading') {
+    return (
+      <div className="relative">
+        <LoadingIndicator text="Loading Datasets..."></LoadingIndicator>;
+      </div>
+    );
+  }
+
+  // Empty datasets view
+  if (loadingState === 'completed' && datasets.length === 0) {
+    return (
+      <div className="space-y-4">
+        <h2 className="text-2xl font-semibold">Select a Dataset</h2>
+        <p className="">You have not created any Datasets for this account!</p>
+        <p>
+          To create a new dataset, click{' '}
+          <span className="text-accent">
+            {' '}
+            <Link href="/datasets/create">here</Link>
```

Comment:
I don't think `/datasets/create` exists? Should this be `/datasets/new`?

Reply:
Great catch will change in next revision